### ISSUE TITLE
`ts/indent` Problems with the indent rule

### ISSUE THREAD

**Issue by Namchee** [eslint-stylistic/eslint-stylistic#126]
Mirror of https://github.com/typescript-eslint/typescript-eslint/issues/1824

TL;DR - the `indent` rule is broken on TypeScript codebases. Due to bandwidth constraints it will remain broken _**indefinitely**_ unless the community is able to step up to fix it. Many users are working around it via ESLint disable comments, and configuring certain nodes as ignored; both of which are poor experiences.

**Comment by antfu** [comment]
I believe this rule soles most of the cases: https://github.com/antfu/eslint-plugin-antfu/blob/main/src/rules/indent-binary-ops.md

We will consider move it to ESLint Stylistic (#136)

**Comment by thecrow** [comment]
The `indent` rule seems too wordy in scope, it makes the rule difficult to code, test, and affects performance. I think it would be good to separate the `indent` rule into smaller rules, like the [`indent-binary-ops`](https://github.com/antfu/eslint-plugin-antfu/blob/b7399c27b277e91ae69e3654906ead405ff31294/src/rules/indent-binary-ops.md) plugin.

**Comment by finscn** [comment]
It can't work correctly after  PropertyDefinition : 
<img width="314" alt="image" src="https://github.com/eslint-stylistic/eslint-stylistic/assets/288367/f0e28537-95ef-4352-b255-f055099bf722">
